java/src/jmri/jmrix/mqtt/MqttThrottle.java

```diff
@@ -0,0 +1,285 @@
+
+        finishRecord();
+    }
+
+
+
+    public int setDccAddress(int newaddress) {
+
+        if (address > 0) {
+            // Send blank MQTT message to remove any persistent message
+            jmri.util.ThreadingUtil.runOnLayout(() -> {
+                mqttAdapter.publish(this.sendThrottleTopic.replaceFirst("\\$address", String.valueOf(address)), "");
+                mqttAdapter.publish(this.sendDirectionTopic.replaceFirst("\\$address", String.valueOf(address)), "");
+
+                for (int functionNum = 0; functionNum < getFunctions().length; functionNum++) {
+                    mqttAdapter.publish(this.sendFunctionTopic.replaceFirst("\\$address",
+                        String.valueOf(address)).replaceFirst("\\$function",String.valueOf(functionNum)), "");
+                }
+            });
+
+            mqttAdapter.unsubscribe(this.rcvThrottleTopic.replaceFirst("\\$address", String.valueOf(address)), this);
+            mqttAdapter.unsubscribe(this.rcvDirectionTopic.replaceFirst("\\$address", String.valueOf(address)), this);
+            mqttAdapter.unsubscribe(this.rcvFunctionTopic.replaceFirst("\\$address", String.valueOf(address)).replaceFirst("\\$function", "#"), this);
+        }
+        address = newaddress;
+
+        mqttAdapter.subscribe(this.rcvThrottleTopic.replaceFirst("\\$address", String.valueOf(address)), this);
+        mqttAdapter.subscribe(this.rcvDirectionTopic.replaceFirst("\\$address", String.valueOf(address)), this);
+        mqttAdapter.subscribe(this.rcvFunctionTopic.replaceFirst("\\$address", String.valueOf(address)).replaceFirst("\\$function", "#"), this);
+
+        consistManager.activateConsist(getLocoAddress());
+        setSpeedSetting(0);
+        setIsForward(true);
+
+        functionPattern = Pattern.compile(this.rcvFunctionTopic.replaceFirst("\\$address",
+            String.valueOf(address)).replaceFirst("\\$function", "(\\\\d+)"));
+
+        return address;
+    }
+
+    public int getDccAddress() {
+        return address;
+    }
+
+    @Override
+    public LocoAddress getLocoAddress() {
+        return new DccLocoAddress(address, MqttThrottleManager.isLongAddress(address));
+    }
+
+    @Override
+    public void notifyMqttMessage(String receivedTopic, String message) {
+
+        if (receivedTopic.endsWith(this.rcvThrottleTopic.replaceFirst("\\$address", String.valueOf(address)))) {
+            super.setSpeedSetting(Math.max(0.0f,Math.min(Float.parseFloat(message)/100.0f,1.0f)));
+
+        } else if (receivedTopic.endsWith(this.rcvDirectionTopic.replaceFirst("\\$address",
+                    String.valueOf(address)))) {
+            switch (message) {
+                case "FORWARD":
+                    super.setIsForward(true);
+                    break;
+                case "REVERSE":
+                    super.setIsForward(false);
+                    break;
+                case "STOP":
+                    super.setSpeedSetting(-1);
+            }
```

Comment:
Static analysis failes on this line due to that this switch case statement should have a default case. If the default case should never happen, do a log.error().

Reply:
Fixed